### PR TITLE
[CS-4184]: Fixes android auth issues

### PR DESCRIPTION
**Pull request by danibonilha**
### Description

Fixes the payment link, and authentication loop on Android 10 , unfortunately it reintroduces the permission locking issue, but better be safe than sorry, and it only happens one per app install.

### Checklist

- [x] Tested on iOS
- [x] Tested on Android

### Screenshots

https://user-images.githubusercontent.com/20520102/177630633-a2b2ea9f-112d-4c89-a0b4-595b3b38f429.mp4

### CODE REVIEW

File: cardstack/src/hooks/useAppState.ts [hunk 1]
```diff
@@ -39,11 +39,17 @@ export const useAppState = () => {
 
   const isActive = useMemo(() => appState === AppStateType.active, [appState]);
 
+  const isActiveWithoutUpdate = useMemo(
+    () => AppState.currentState === AppStateType.active,
```
Comment (oidoug) [inline cardstack/src/hooks/useAppState.ts]: So if I understand this correctly, `isActiveWithoutUpdate` will be true on app open and then never change during app's life time, because there's no reactive dependency here and this hook isn't unmounted on app going to background.

Is that intentional? What is its purpose?

File: cardstack/src/redux/authSlice.ts [hunk 2]
```diff
@@ -58,11 +66,21 @@ export const useAuthActions = () => {
...
   }, [dispatch]);
 
+  const authWithBiometricsStarted = useCallback(() => {
+    dispatch(authSlice.actions.authWithBiometricsStarted());
+  }, [dispatch]);
+
+  const authWithBiometricsFinished = useCallback(() => {
+    dispatch(authSlice.actions.authWithBiometricsFinished());
+  }, [dispatch]);
+
```
Comment (oidoug) [inline cardstack/src/redux/authSlice.ts:76]: I saw some posts online recommending this approach but decided on the timer, but looks like this will be more reliable. Might be worth trying to use it for the camera permission issue too.

File: cardstack/src/screens/UnlockScreen/useUnlockScreen.tsx [hunk 3]
```diff
@@ -157,10 +177,10 @@ export const useUnlockScreen = () => {
   }, [checkUserTemporaryBlock, inputPin, validatePin]);
 
   useEffect(() => {
-    if (isBiometryEnabled && isActive) {
+    if (isBiometryEnabled && isActiveWithoutUpdate) {
```
Comment (oidoug) [inline cardstack/src/screens/UnlockScreen/useUnlockScreen.tsx]: Checking for `isActiveWithoutUpdate` here seems like the same as just checking for `isBiometryEnabled` only, I don't think their value change, the effect is called because the whole useUnlockScreen hooks is re-rendered. I understand having it if it's handling a corner case I don't see.

**Review by oidoug — APPROVED** [review]
Good to merge once tests are passing. Just left some comments for minor concerns.

Additionally, I saw the biometric auth once when unlock screen wasn't showing and once more after minimizing the app, but couldn't replicate neither of the cases 🤷
